File: app/src/main/java/com/alexpaxom/workrhythm/TimeFormatter.java
package com.alexpaxom.workrhythm;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.TimeZone;

public class TimeFormatter {
    private static final int convertMillisecondsToDays = 86400000;

    public static String intervalFromTime(Long elapsedTime) {
        DateFormat df = new SimpleDateFormat(" HH:mm:ss");
        df.setTimeZone(TimeZone.getTimeZone("GMT+0"));
        Integer elapsed_days = (int) (elapsedTime / convertMillisecondsToDays);
        return App.getInstance().getString(R.string.time_format_days) + " " + elapsed_days.toString()
                + " " +
                App.getInstance().getString(R.string.time_format_hours) + " " + df.format(elapsedTime%convertMillisecondsToDays);
    }

    public static String intervalGetDays(Long elapsedTime) {
        Integer elapsed_days = (int) (elapsedTime / convertMillisecondsToDays);
        return elapsed_days.toString();
    }

    public static String intervalGetHours(Long elapsedTime) {
        DateFormat df = new SimpleDateFormat(" HH:mm:ss");
        df.setTimeZone(TimeZone.getTimeZone("GMT+0"));

        return df.format(elapsedTime%convertMillisecondsToDays);
    }
}

File: app/src/main/java/com/alexpaxom/workrhythm/helpers/TimeFormatter.kt
package com.alexpaxom.workrhythm.helpers

import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

object TimeFormatter {
    private const val convertMillisecondsToDays = 86400000
    fun intervalFromTime(elapsedTime: Long): String {
        val df: DateFormat = SimpleDateFormat(" HH:mm:ss")
        df.timeZone = TimeZone.getTimeZone("GMT+0")
        val elapsed_days = (elapsedTime / convertMillisecondsToDays).toInt()
        return ("Days: " + elapsed_days.toString()
                + " " +
                "Hours: " + df.format(elapsedTime % convertMillisecondsToDays))
    }

    fun intervalGetDays(elapsedTime: Long): String {
        val elapsed_days = (elapsedTime / convertMillisecondsToDays).toInt()
        return elapsed_days.toString()
    }

    fun intervalGetHours(elapsedTime: Long): String {
        val df: DateFormat = SimpleDateFormat(" HH:mm:ss")
        df.timeZone = TimeZone.getTimeZone("GMT+0")
        return df.format(elapsedTime % convertMillisecondsToDays)
    }
}